### ISSUE TITLE
Fluent-Bit 1.8.12 - Ubuntu APT Package TLS Issue

### ISSUE THREAD

**Issue by joeyberkovitz** [fluent/fluent-bit#4724]
## Bug Report

**Describe the bug**

As originally reported under https://github.com/fluent/fluent-bit/issues/4085, it has been discovered that the official Ubuntu package of Fluent-Bit wasn't properly compiled with OpenSSL, and as a result uses mbedtls for TLS. This is problematic because mbedtls doesn't support all TLS certificates, and as a result, crashes when given a certificate that it doesn't support.

The issue was fixed in https://github.com/fluent/fluent-bit-packaging/pull/26, but the latest release of Fluent-Bit - 1.8.12 still uses mbedtls. If a deb file is produced from the fluent-bit-packaging repository with `./build.sh -v 1.18.12 -d ubuntu/20.04`, it works properly, so it seems that whatever build process was used has yet to be updated.

**To Reproduce**
- Example log message if applicable:
```
[2022/01/31 14:05:57] [error] [lib] backend failed
[2022/01/31 14:05:57] [error] [tls] /tmp/fluent-bit-1.8.12/src/tls/mbedtls.c:178 X509 - The extension tag or value is invalid : ASN1 - ASN1 tag was of a
[2022/01/31 14:05:57] [error] [TLS] Invalid CA file: ...ca_cert.pem
[2022/01/31 14:05:57] [error] [tls] could not create TLS backend
```

**Comment by patrick-stephens** [comment]
I think this has run afoul of what you indicated in that original issue as well: namely that the base images are cached so the updates are not reflected unless you force a rebuild. As a specific build server is being used, it is caching the previous images so we need to force an update for them.

I'll ask @edsiper if we can do a quick 1.8.13 for this issue to get a rebuild out - we could also update the 1.8.12 packages but my concern is that affects people using them now in that it will change underneath them if they're doing any testing, sign off, etc.

This will no longer happen for the 1.9+ builds as all packaging is done via the ephemeral Github runners plus I switched it over to using multi-stage for the builds: https://github.com/fluent/fluent-bit/tree/master/packaging

**Comment by patrick-stephens** [comment]
Yeah as in delete them on the build server, it's the base ones that are the issue.
It won't happen in 1.9 as multistage and ephemeral runners means they'll update

**Comment by joeyberkovitz** [comment]
@patrick-stephens - just checking in on this

**Comment by patrick-stephens** [comment]
Apologies @joeyberkovitz, I was actually meaning to message you yesterday but got distracted by other issues.
I believe a 1.8.13 release is imminent but @edsiper can confirm that - this should have the base images pruned on the build server so they should build from scratch per your original change.

However, one of the things I've been sorting is a regular nightly build and this actually would be a good test of it if you can verify the packages here are ok for you?
https://github.com/fluent/fluent-bit/releases/tag/unstable

I'm still sorting a few things out with the CI for this nightly build but the actual code and binaries there should have the issue resolved - there are other issues like GH intermittently rejecting such a large number of binaries being uploaded.

**To be clear: this build is not for production deployment.** These will go into 1.9 though eventually so be good to confirm we have definitely fixed it for you.

**Comment by joeyberkovitz** [comment]
Question - what's the difference between `packages-master-ubuntu-20.04-td-agent-bit_1.9.0_amd64.deb` and `packages-master-ubuntu-20.04-fluent-bit_1.9.0_amd64.deb`

I tested the 1.9.0 td-agent-bit on one of my Ubuntu 20.04 machines and verified that it starts up properly and a TLS dependent pipeline works properly.

**Comment by patrick-stephens** [comment]
Essentially not a huge amount other than naming (although that affects where config files, etc. live). The difference is driven via the `FLB_TD=On|Off` setting to cmake.

For 1.9 we're deprecating `td-agent-bit` and moving to `fluent-bit` but providing both to allow migration: https://docs.fluentbit.io/manual/v/1.9-pre/installation/linux#migration-to-fluent-bit

**Comment by edsiper** [comment]
NOTE: 1.9 has not been officially released.

**Comment by patrick-stephens** [comment]
@joeyberkovitz 1.8.13 should be out now with this resolved. Can you check it is ok?
